primitiveFTPd/src/org/primftpd/services/QuickSettingsService.java

```diff
@@ -29,6 +29,30 @@ public void onCreate() {
     public void onClick() {
         logger.debug("onClick");
         super.onClick();
+        boolean unlockedOnly = false; // Grab value from somewhere
+
+        if(unlockedOnly) {
+            // Check whether the device is locked or not.
+            if(isLocked()){
```

Comment:
These two if blocks could be put together like "if (unlockedOnly && isLocked())". This would avoid one "else { toggle() }" block